src/test/java/plugins/PipelineTest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2023 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package plugins;
+
+import org.apache.commons.lang3.SystemUtils;
+import org.jenkinsci.test.acceptance.AbstractPipelineTest;
+import org.jenkinsci.test.acceptance.junit.WithPlugins;
+import org.jenkinsci.test.acceptance.plugins.maven.MavenInstallation;
+import org.jenkinsci.test.acceptance.po.Build;
+import org.jenkinsci.test.acceptance.po.WorkflowJob;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.WebElement;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.jenkinsci.test.acceptance.Matchers.hasContent;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Tests a complete pipeline flow:
+ * 1- Checkout repo from github
+ * 2- Execute tests
+ * 3- Generate javadoc
+ * 4- Assert results have been collected
+ *
+ */
+@WithPlugins({"git", "junit", "javadoc"})
+public class PipelineTest extends AbstractPipelineTest {
+
+    @Before
+    public void setup() {
+        MavenInstallation.installMaven(jenkins, "M3", "3.3.9");
+    }
+
+    @Test
+    public void testCompletePipeline() {
```

Comment:
whilst it appears as though the functionality is covered in [`WorkflowPluginTest#linearFlow()`](https://github.com/jenkinsci/acceptance-test-harness/blob/3bb860d86858dd44d7500e3b51776dc28e408f22/src/test/java/plugins/WorkflowPluginTest.java#L122-L175} the test is subtly different and is exercising a useful path of a build with failing tests (where the failing tests are recorded and checked)